### PR TITLE
ClientConfiguration endpoint added

### PR DESCRIPTION
**Pull request by MorfMichael**
added an api controller for the client configurations

### CODE REVIEW

File: src/Codeworx.Identity.EntityFrameworkCore.Api/ClientConfigurationController.cs [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+        }
+
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<ClientConfigurationInfoData> InsertClientConfigurationAsync([FromBody] ClientConfigurationInsertData configuration)
+        {
+            var entity = new ClientConfiguration
+            {
+                Id = Guid.NewGuid(),
+                ClientSecretHash = _hashingProvider.Create(configuration.Secret),
```
Comment (schwarzr) [inline src/Codeworx.Identity.EntityFrameworkCore.Api/ClientConfigurationController.cs:110]: The Client Secret shoud be server generated and returned from the Post call.

Also a HttpPut ResetClientSecret Operation is required.

Only generate clientSecrets for clientTypes  (Web = 0x01,   Backend = 0x08 and  ApiKey = 0x10)

File: src/Codeworx.Identity.EntityFrameworkCore.Api/ClientConfigurationController.cs [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+        {
+            var entity = await _db.Context.Set<ClientConfiguration>().FirstOrDefaultAsync(x => x.Id == configuration.Id);
+
+            if (entity == null)
+            {
+                // TODO return 404
+                throw new NotImplementedException();
+            }
+
+            entity.AccessTokenType = configuration.AccessTokenType;
```
Comment (schwarzr) [inline src/Codeworx.Identity.EntityFrameworkCore.Api/ClientConfigurationController.cs:138]: add HasClientSecret to response

File: src/Codeworx.Identity.EntityFrameworkCore.Api/ClientConfigurationController.cs [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+                throw new NotSupportedException();
+            }
+
+            return redirectUrl;
+        }
+
+        [HttpPost("{id}/redirect-urls")]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<IEnumerable<ValidRedirectUrlInfoData>> InsertValidRedirectUrlsAsync(Guid id, [FromBody] IEnumerable<string> urls)
```
Comment (schwarzr) [inline src/Codeworx.Identity.EntityFrameworkCore.Api/ClientConfigurationController.cs:218]: Insert only one Url at a time (consistency with delete operation)

File: src/Codeworx.Identity.EntityFrameworkCore.Api/Model/ClientConfigurationInfoData.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+        public string AccessTokenType { get; set; }
+
+        public string AccessTokenTypeConfiguration { get; set; }
+
+        public AuthenticationMode AuthenticationMode { get; set; }
+
+        public ClientType ClientType { get; set; }
+
+        public Guid? UserId { get; set; }
```
Comment (schwarzr) [inline src/Codeworx.Identity.EntityFrameworkCore.Api/Model/ClientConfigurationInfoData.cs:22]: simpel User Object

File: src/Codeworx.Identity.EntityFrameworkCore.Api/Model/ClientConfigurationInfoData.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+        public List<Guid> Scopes { get; set; }
```
Comment (schwarzr) [inline src/Codeworx.Identity.EntityFrameworkCore.Api/Model/ClientConfigurationInfoData.cs:24]: equivalent to the user property, use a simple Scope opbject with id and displayName..

**Review by schwarzr — CHANGES_REQUESTED** [review]
Some minor changes neede... See comments.

File: src/Codeworx.Identity.EntityFrameworkCore.Api/Model/ClientConfigurationInsertData.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+        public string AccessTokenType { get; set; }
+
+        public string AccessTokenTypeConfiguration { get; set; }
+
+        public AuthenticationMode AuthenticationMode { get; set; }
+
+        public ClientType ClientType { get; set; }
+
+        public Guid? UserId { get; set; }
```
[the remark this answers is not in the document]
Comment (AEonWho) [reply src/Codeworx.Identity.EntityFrameworkCore.Api/Model/ClientConfigurationInsertData.cs]: done

File: src/Codeworx.Identity.EntityFrameworkCore.Api/ClientConfigurationController.cs (hunk 1 above)
Comment (AEonWho) [reply src/Codeworx.Identity.EntityFrameworkCore.Api/ClientConfigurationController.cs]: done

File: src/Codeworx.Identity.EntityFrameworkCore.Api/ClientConfigurationController.cs (hunk 2 above)
Comment (AEonWho) [reply src/Codeworx.Identity.EntityFrameworkCore.Api/ClientConfigurationController.cs]: done

File: src/Codeworx.Identity.EntityFrameworkCore.Api/ClientConfigurationController.cs (hunk 3 above)
Comment (AEonWho) [reply src/Codeworx.Identity.EntityFrameworkCore.Api/ClientConfigurationController.cs]: done

File: src/Codeworx.Identity.EntityFrameworkCore.Api/Model/ClientConfigurationInfoData.cs (hunk 5 above)
Comment (AEonWho) [reply src/Codeworx.Identity.EntityFrameworkCore.Api/Model/ClientConfigurationInfoData.cs]: done

File: src/Codeworx.Identity.EntityFrameworkCore.Api/Model/ClientConfigurationInfoData.cs (hunk 4 above)
Comment (AEonWho) [reply src/Codeworx.Identity.EntityFrameworkCore.Api/Model/ClientConfigurationInfoData.cs]: done

File: src/Codeworx.Identity.EntityFrameworkCore.Api/Extensions/ClientConfigurationOptions.cs
Comment (schwarzr) [inline src/Codeworx.Identity.EntityFrameworkCore.Api/Extensions/ClientConfigurationOptions.cs:1]: Namespace looks wrong...

**Review by schwarzr — CHANGES_REQUESTED** [review]

File: src/Codeworx.Identity.EntityFrameworkCore.Api/Extensions/ClientConfigurationOptions.cs
Comment (AEonWho) [reply src/Codeworx.Identity.EntityFrameworkCore.Api/Extensions/ClientConfigurationOptions.cs:1]: Fixed

**Review by schwarzr — APPROVED** [review]

**Pull request merged**